src/tike/ptycho/ptycho.py

```diff
@@ -494,19 +504,21 @@ def iterate(self, num_iter: int) -> None:
                     self.parameters.probe,
                     self.parameters.object_options.preconditioner,
                 )))
+            # ELSE IF: self.parameters.probe_options.rescale_using_fixed_probe_intensity_photons
+            # ELSE:    do nothing
+ 
+            if self.parameters.probe_options.recover_probe :
+                if self.parameters.eigen_probe is not None:
```

Comment:
Use a logical "and" instead of adding an indentation level here because there is only one function call in these blocks.